src/tequila/circuit/visualize.py

```diff
@@ -0,0 +1,156 @@
+              file_path: Optional[str] = None) -> figure.Figure:
+    """
+    Precondition:
+    The maximum number of qubits is 10 at the moment (Feb24, 2023)
+
+    Post condition:
+    The graph of the qubit_hamiltonian is displayed
+    or is stored in file_path
+
+    One thing to note is that if you are using command-line interface,
+    the plot might not be successfully shown
+
+    , so it's better idea to save it as an image.
+    === parameters ===
+    qubit_hamiltonian: A QubitHamiltonian representation of pauli operators
+    circuit: A QCircuit that corresponds to the
+    connectivity: networkx.Graph.edges that show the connectivity of qubits
+    A nested list should be provided for this argument.
+    file_name: str format of file name
+    to which the plotted graph will be exported. The file will be a png format
+
+    === return ===
+    None
+
+    === sample usages ===
+    >>> visualize(
+    >>> tequila.QubitHamiltonian(openfermion.QubitOperator("X0 X5 Y3", 1)))
+    *** A graph with nodes 0 and 5 having colour red
+    and node 3 having colour green ***
+    >>> visualize(
+    >>> tequila.QubitHamiltonian(openfermion.QubitOperator("X0 X5 Y3", 1)),
+    >>> circuit=gates.X(0) + gates.CNOT(0, 5) + gates.Y(3))
+    *** A graph with nodes 0 and 5 having color red and
+    node 3 having colour green with edge 0 and 5 exists ***
+    >>> visualize(
+    >>> tequila.QubitHamiltonian(openfermion.QubitOperator("X0 X5 Y3", 1)),
+    >>> connectivity=[(0, 0), (0, 5), (3, 3)])
+    *** A graph with nodes 0 and 5 having color red and
+    node 3 having colour green with edge 0 and 5 exists ***
+    >>> visualize(
+    >>> tequila.QubitHamiltonian(openfermion.QubitOperator("X0 X5 Y3", 1)),
+    >>> connectivity=[(0, 0), (0, 5), (3, 3)],
+    >>> file_name="test_system")
+    *** Exported an image of a graph with nodes 0 and 5 having color red and
+    node 3 having colour green with edge 0 and 5 exists to test_system.png ***
+    """
+    if(len(qubit_hamiltonian.qubit_operator.terms)) != 1:
+        raise Exception("The input qubit_operator should have length 1")
+
+    qh = next(iter(qubit_hamiltonian.qubit_operator.terms))
+    graph = networkx.Graph()
+    graph, pos = _draw_basics(graph)
+    if circuit is None and connectivity is None:
+        graph, pos = _visualize_helper(qh, graph, pos)
+
+    elif connectivity is None:
+        lConn = _circuit_to_connectivity(circuit)
+        graph.add_edges_from(lConn)
+        graph, pos = _visualize_helper(qh, graph, pos, lConn)
+
+    elif circuit is None:
+        graph, pos = _visualize_helper(qh, graph, pos, connectivity)
+
+    if file_path is None:
+        return plt.figure()
+    if file_path is not None:
+        plt.savefig(file_path+".png", format="PNG")
+        return plt.figure()
+
+
+def _circuit_to_connectivity(circuit):
+    """
+    A helper function for visualize() function.
+    This function returns the list of connections found in circuit
+    """
```

Comment:
can you try to use: circuit.to_networkx() for this?
This gives you back a networkx graph that should have the collected connections as edges